guide/api-design-guidelines.md

```diff
@@ -0,0 +1,181 @@
+
+<!-- markdownlint-disable MD033 MD049 MD055 -->
+<!--
+Note to contributors: All guidelines have an anchor tag to allow cross-referencing from associated tooling.
+The anchor tags within a section using a common prefix to ensure uniqueness with anchor tags in other sections.
+Please ensure that you add an anchor tag to any new guidelines that you add and maintain the naming convention.
+-->
+
+## History
+
+<details>
+  <summary>Expand change history</summary>
+
+| Date        | Notes                                                          |
+| ----------- | -------------------------------------------------------------- |
+| 2024-Sep-17 | Added guidance on Secrets                                      |
+
+</details>
+
+## Introduction
+
+These are prescriptive guidelines that Radius contributors MUST follow while designing APIs to ensure that customers have a great experience. These guidelines help make Radius APIs:
+- Developer friendly via consistent patterns
+- Efficient & cost-effective
+
+Technology and software is constantly changing and evolving, and as such, this is intended to be a living document. [Open an issue](https://github.com/radius-project/design-notes/issues) to suggest a change or propose a new idea. 
+
+### Prescriptive Guidance
+This document offers prescriptive guidance labeled as follows:
+
+:white_check_mark: **DO** adopt this pattern. If you feel you need an exception, present your reason at a design review discussion **prior** to implementation.
+
+:ballot_box_with_check: **YOU SHOULD** adopt this pattern. If not following this advice, you MUST disclose your reason during a design review discussion.
+
+:heavy_check_mark: **YOU MAY** consider this pattern if appropriate to your situation.
+
+:warning: **YOU SHOULD NOT** adopt this pattern. If not following this advice, you MUST disclose your reason during a design review discussion.
+
+:no_entry: **DO NOT** adopt this pattern. If you feel you need an exception, present your reason at a design review discussion **prior** to implementation.
+
+## API Foundation:
+
+Radius provides an HTTP-based API that deploys and manages cloud-native applications as well as on-premise or cloud resources.
+The overall design of the Radius API is based on the Azure Resource Manager API (ARM). Radius generalizes the design of ARM, removes proprietary Azure concepts, and extends the ARM contract to support the conventions of other resource managers like Kubernetes or AWS Cloud-Control. 
+
+Radius utilizes TypeSpec for describing API model definitions and operations, providing a consistent approach to defining the structure of API payloads and interactions. The TypeSpec specification serves as the source to generate corresponding OpenAPI 2.0 (swagger) API documentation.
+
+Reference: 
+- [Radius API](https://docs.radapp.io/concepts/technical/api/)
+- [Radius Architecture](https://docs.radapp.io/concepts/technical/architecture/)
+
+## Guidelines
+
+### Resource Property Design
```

Comment:
These are all secret-related. Is the expectation that this file will be updated as more API patterns are used?